dwave/cloud/client.py

```diff
@@ -1195,6 +1211,14 @@ def _handle_problem_status(self, message, future):
 
                 # If the message is complete, forward it to the future object
                 if 'answer' in message:
+
+                    # If the future does not know which solver it's associated
+                    # with, we get it from the info provided from the server.
+                    # An alternative to making this call here would be to pass
+                    # self in with the message
+                    if future.solver is None:
```

Comment:
I am not sure how much of a hack it is to do this here. As I mention in the comment, an alternative would be to do something like
```
future._set_message_and_solver(message, client)
```
Thoughts?

Reply:
This whole function is a hack, and I want to refactor it at some point, having in mind we have better control over problem/answer/metadata access on SAPI since recently.